### DISCUSSION TITLE
web components UI library planned?

### DISCUSSION THREAD

**Discussion by deckerdent** [primefaces/community#1356] [PrimeNG] [answered]
Hi, first of all, I know it's not the right discussion category, sorry for that. There was no suitable category and I tried other ways first, I promise. 

Is a plain web components library for prime planned? Our company uses PrimeFaces, PrimeNG, PrimeVue AND PrimeReact in different projects at the moment. PrimeFaces only in legacy projects, but the others pretty actively, especially PrimeNg. We are now planning to start with micro-frontends and allow teams to choose their frontend-framework project by project, which is why we want framework agnostic components rather than multiple deps for the same purpose. 

This gives us a hard time as we would really love to continue to use prime as the foundation of our platform. And we wondered if prime would not work on such a lib.

Is something planned in this direction? I couldn't find any information about it, but a co-worker mentioned they "think they've read about it will come this year".... I shortly searched the discussions here, the blog and even the roadmaps.

**Comment by nckirik** [comment]
@cagataycivici mentioned(*) they had plans for web components in 2022, but i guess those plans were postponed since primevue took priority over web components.

(*) https://twitter.com/cagataycivici/status/1541697892410327045

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Yes, we have plans to make web components. Right now the team is focused on other features for Prime UI Libraries.

**Reply by deckerdent** [reply]
~~Just saw the first bite is available! PrimeElements, that's amazing!~~

Just saw what I found seems to be a relic https://www.primefaces.org/primeui/#primeelements seems to be abandoned since 2016. You should maybe hide it away from robots so it does not get indexed or take it down for the moment as I guess it's no officially supported thing.
